src/d3a/models/strategy/load_hours.py

```diff
@@ -160,27 +175,44 @@ def _delete_past_state(self):
 
     def _area_reconfigure_prices(self, **kwargs):
         if key_in_dict_and_not_none(kwargs, 'initial_buying_rate'):
-            self.bid_update.initial_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY,
-                                                                  kwargs['initial_buying_rate'])
+            initial_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY,
+                                                  kwargs['initial_buying_rate'])
+        else:
+            initial_rate = self.bid_update.initial_rate
         if key_in_dict_and_not_none(kwargs, 'final_buying_rate'):
-            self.bid_update.final_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY,
-                                                                kwargs['final_buying_rate'])
+            final_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY,
+                                                kwargs['final_buying_rate'])
+        else:
+            final_rate = self.bid_update.final_rate
         if key_in_dict_and_not_none(kwargs, 'energy_rate_increase_per_update'):
-            self.bid_update.energy_rate_change_per_update = \
+            energy_rate_change_per_update = \
                 read_arbitrary_profile(InputProfileTypes.IDENTITY,
                                        kwargs['energy_rate_increase_per_update'])
+
+        else:
+            energy_rate_change_per_update = self.bid_update.energy_rate_change_per_update
         if key_in_dict_and_not_none(kwargs, 'fit_to_limit'):
-            self.bid_update.fit_to_limit = kwargs['fit_to_limit']
+            fit_to_limit = kwargs['fit_to_limit']
+        else:
+            fit_to_limit = self.bid_update.fit_to_limit
         if key_in_dict_and_not_none(kwargs, 'update_interval'):
             if isinstance(kwargs['update_interval'], int):
                 update_interval = duration(minutes=kwargs['update_interval'])
             else:
                 update_interval = kwargs['update_interval']
-            self.bid_update.update_interval = update_interval
+        else:
+            update_interval = self.bid_update.update_interval
+
         if key_in_dict_and_not_none(kwargs, 'use_market_maker_rate'):
             self.use_market_maker_rate = kwargs['use_market_maker_rate']
 
-        self._validate_rates()
+        if self._validate_rates(initial_rate, final_rate, energy_rate_change_per_update,
```

Comment:
I don't like this approach of using validator method for logic testing, but can't quickly think of any alternate and better way either :-(

Reply:
To be frank, I think I prefer this than the reactive approach before. This way all event parameters are either discarded or applied. There is no way that, for instance, only initial_rate and final_rate to be applied , and for energy_rate_change_per_update and fit_to_limit to not be applied.